invenio_pidstore/models.py

```diff
@@ -311,7 +315,7 @@ def unassign(self):
                 self.object_uuid = None
                 db.session.add(self)
         except SQLAlchemyError:
-            logger.exception("Failed to unassign object.".format(self),
+            logger.exception("Failed to unassign object.",
```

Comment:
@drjova it looks to me that this is the only fix?
In general, why do you prefer `%s` over `format`? Why is this change needed?
And why did you keep the last with `format` then?

Just need a little bit more context :) Thanks!

Reply:
Well it's a very common issue, let's say you've sentry as a log aggregator. If you call the same log statement twice without interpolation ```logger.exception('Failed to assign %s:%s', 'rec', '111')``` and ```logger.exception('Failed to assign %s:%s', 'rec', '222')```, then both exceptions will be grouped as two occurrences of the same log message, with different values for ``%s:%s``.

With interpolation (i.e. ```format```) , you'll get one instance of two different log messages, which is less useful.

> And why did you keep the last with format then?

WDYM :)

Reply:
Thanks a lot for the explanation! Now it is more clear.
I would the suggest to highlight this somehow: the risk is that next PR, someone will come and convert all of them to `format`! :)
I don't have better ideas than a comment or a well named function/method to wrap the `logger.exception`...

Reply:
👍 TBH I'd be in favour to just raise and remove the log exceptions and let the application to log it but I guess this is a bigger change. WDYT?

Reply:
or another idea will be to make them ``error``

Reply:
It saw this too in the past, what actually happens when using `%s` is that it exposes the message template to log handlers rather than the formatted string, formatting is delayed until it is necessary, which in turn makes the two examples @drjova wrote look the same for things like Sentry.

If you check the documentation, [PEP 282](https://www.python.org/dev/peps/pep-0282/) and [docs](https://docs.python.org/2/library/logging.html) they actually tell us to use `%s`, if we don't like them the logger can be configured to use `{}` or `$`, but I wouldn't go down this way.

Reply:
Shall we completely remove the ``logger.exception`` anyhow the exception will be raised hence we don't need to duplicate it

Reply:
I'd be in favor of that, but we might be losing valuable information, i.e. the message, which will be hidden inside the default exception and probably more difficult to find.

What about using custom exceptions for this? (I guess this would qualify as an API change since we are changing the exception that will be handled later on)

Reply:
Yes, I think we can remove the logging of errors. It's primarily IntegrityError which will be masked, but that should be ok.

Reply:
It would require new minor release, and could be shipped in Invenio v3.3.